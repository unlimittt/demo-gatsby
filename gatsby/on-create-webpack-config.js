const onCreateWebpackConfig = ({ stage, actions }) => {
  if (stage.startsWith("develop")) {
    actions.setWebpackConfig({
      resolve: {
        alias: {
          "react-dom": "@hot-loader/react-dom",
        },
      },
    })
  }
}

module.exports = onCreateWebpackConfig
