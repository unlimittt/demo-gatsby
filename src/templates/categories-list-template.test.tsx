import React from 'react';
import { useStaticQuery, StaticQuery } from 'gatsby';
import CategoriesListTemplate from './categories-list-template';
import siteMetadata from '../../jest/__fixtures__/site-metadata';
import allMarkdownRemark from '../../jest/__fixtures__/all-markdown-remark';
import type { RenderCallback } from '../types';
import Enzyme, { shallow } from "enzyme"
import Adapter from "enzyme-adapter-react-16"

Enzyme.configure({ adapter: new Adapter() })

describe('CategoriesListTemplate', () => {
  const props = {
    ...siteMetadata,
    ...allMarkdownRemark
  };

  beforeEach(() => {
    (StaticQuery as jest.MockedFunction<typeof StaticQuery | any>).mockImplementationOnce(
      ({ render }: RenderCallback) => (
        render(props)
      ),
      (useStaticQuery as jest.MockedFunction<typeof useStaticQuery | any>).mockReturnValue(props)
    );
  });

  it('renders correctly', () => {
    const wrapper = shallow(<CategoriesListTemplate />);
    expect(wrapper).toMatchSnapshot();
  });
});
