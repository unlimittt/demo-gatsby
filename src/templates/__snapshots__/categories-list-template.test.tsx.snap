// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoriesListTemplate renders correctly 1`] = `
<Layout
  description="Test subtitle"
  title="Categories"
>
  <Sidebar />
  <Page
    title="Categories"
  >
    <ul>
      <li
        key="test_0"
      >
        <mockConstructor
          to="/category/test-0/"
        >
          test_0
           (
          1
          )
        </mockConstructor>
      </li>
      <li
        key="test_1"
      >
        <mockConstructor
          to="/category/test-1/"
        >
          test_1
           (
          2
          )
        </mockConstructor>
      </li>
    </ul>
  </Page>
</Layout>
`;
